packages/strapi-plugin-i18n/admin/src/index.js

```diff
@@ -1,28 +1,68 @@
+import * as yup from 'yup';
 import pluginPkg from '../../package.json';
+import middlewares from './middlewares';
 import pluginId from './pluginId';
 import pluginLogo from './assets/images/logo.svg';
 import trads from './translations';
 
 export default strapi => {
   const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;
 
+  middlewares.forEach(middleware => {
+    strapi.middlewares.add(middleware);
+  });
+
   const plugin = {
-    blockerComponent: null,
-    blockerComponentProps: {},
     description: pluginDescription,
     icon: pluginPkg.strapi.icon,
     id: pluginId,
     isReady: true,
-    initializer: () => null,
-    injectedComponents: [],
     isRequired: pluginPkg.strapi.required || false,
-    layout: null,
-    lifecycles: () => {},
     mainComponent: null,
     name: pluginPkg.strapi.name,
     pluginLogo,
-    preventComponentRendering: false,
     trads,
+    boot(app) {
+      const ctbPlugin = app.getPlugin('content-type-builder');
+      const ctbForms = ctbPlugin.internals.forms;
+
+      if (ctbPlugin) {
+        ctbForms.components.add({ id: 'localesPicker', component: () => 'locale picker' });
+
+        ctbForms.extendContentType({
+          validator: {
+            i18n: yup.bool().required(),
+          },
+          form: {
+            advanced() {
```

Comment:
What is the difference between `advanced` and the other one?

Reply:
Advanced and base are the name of each tab of the modal in the content type builder.